File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:

        def dfs(node, maxVal):
            # Base case
            if not node:
                return 0
            
            # increment the count everytime the node value is the biggest
            # incl. edge case where node val == max value (roots must be counted)
            if node.val >= maxVal:
                res = 1
            # Do nothing is node val is smaller
            else:
                res = 0
            
            # Update the new maxVal if node value is bigger
            maxVal = max(maxVal, node.val)

            # Recursive cases
            res += dfs(node.left, maxVal)
            res += dfs(node.right, maxVal)

            return res
        
        return dfs(root, root.val)

        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
#include <algorithm>

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {

        int n = nums.size();
        int left = 0;
        int right = 0; 

             
        
        if (n == 0)
            return 0;

        sort(nums.begin(), nums.end());

        int count = 1;
        while (left < n) {
            vector<int> sequence {nums[left]};
            while (right < n) {
                if (nums[right] == sequence[sequence.size() - 1] + 1)
                    sequence.push_back(nums[right]);
                right++;
            }
            int length = sequence.size();
            cout << length << endl;
            count = max(count, length);
            
            left++;
            right = left;
        }

        return count;
    }

};
